src/cpp/src/sampler.hpp

```diff
@@ -40,6 +41,13 @@ struct SamplerOutput {
     std::unordered_map<uint64_t, std::list<uint64_t>> m_forked_sequences;
 };
 
+struct SequenceGroupSamplingInfo {
+    SamplerOutput sampler_output;
+    size_t max_removed_tokens_per_request = 0; 
```

Comment:
`max_removed_tokens_per_request`, `min_generated_len` and `updated_validation_len` are output of assisting pipelines as `speculative decoding` and `prompt lookup decoding`. We can create one more structure to place them and set in sampling just in case of `assisting pipelines` using